packages/eslint-plugin/src/rules/switch-exhaustiveness-check.ts

```diff
@@ -36,6 +36,13 @@ type Options = [
      * @default false
      */
     requireDefaultForNonUnion?: boolean;
+
+    /**
+     * If `true`, the `default` clause is used to determine whether the switch statement is Exhaustive for union type.
+     *
+     * @default true
+     */
+    allowDefaultCaseMatchUnionMember?: boolean;
```

Comment:
[Refactor] I'm not a fan of truthy defaults for unspecified values. The truthiness/not-defined-ness gets confusing. Proposal: let's have this be falsy by default?

```suggestion
    requireDefaultCaseForUnions?: boolean;
```

Also, tip: _"member"_ refers to methods/properties on a class or object. _"constituent"_ refers to the types in an intersection or union type.